crates/aptos-speculative-state-helper/src/tests/logging.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{SpeculativeEvent, SpeculativeEvents};
+use claims::{assert_err, assert_ok};
+use std::{
+    collections::HashSet,
+    sync::mpsc::{sync_channel, SyncSender},
+};
+
+// Test speculative logging use-case and provide an example implementation.
+
+// Fake local logging levels for testing.
+enum Level {
+    Error = 0,
+    Warn,
+    Debug,
+    Trace,
+}
+
+// Struct to capture speculative logging event.
+struct SpeculativeLog {
+    level: Level,
+    message: String,
+    sender: SyncSender<String>,
+}
+
+impl SpeculativeLog {
+    fn new(level: Level, message: String, sender: SyncSender<String>) -> Self {
+        Self {
+            level,
+            message,
+            sender,
+        }
+    }
+}
+
+// Implementing the required SpeculativeEvents trait. In the real use-case, match
+// would dispatch to real logging macros, but here we send on a channel to test.
+impl SpeculativeEvent for SpeculativeLog {
+    fn dispatch(self) {
+        match self.level {
+            Level::Debug => {
+                self.sender.send(format!("Debug {}", self.message)).unwrap();
```

Comment:
we should probably just ignore the errors instead of unwrap

Reply:
this is in a test, just fake channel for simulating logging.